vspace/src/main/java/edu/asu/diging/vspace/core/aspects/ExhibitionDataAspect.java

```diff
@@ -29,17 +54,86 @@ public void setExhibition(JoinPoint jp) {
         Class<?> returnType = signature.getReturnType();
         // only if there is a model object injected and if we're returning a string
         // (assuming that returning a string implies rendering of a view afterwards
-        if (args != null && returnType == String.class) {
-            for (Object obj : args) {
-                if (obj instanceof Model && !(obj instanceof RedirectAttributes)) {
-                    if (!((Model) obj).containsAttribute("exhibition")) {
+        if(args!=null && returnType==String.class) {
+            for(Object obj : args) {
+                if(obj instanceof Model && !(obj instanceof RedirectAttributes)) {
+                    if(!((Model) obj).containsAttribute("exhibition")) {
                         ((Model) obj).addAttribute("exhibition", exhibitionManager.getStartExhibition());
                     }
-                    if (!((Model) obj).containsAttribute("allSpaces")) {
+                    if(!((Model) obj).containsAttribute("allSpaces")) {
                         ((Model) obj).addAttribute("allSpaces", spaceManager.getAllSpaces());
                     }
                 }
+            }  
+        }
+    }
+
+    @Around("execution(public * edu.asu.diging.vspace.web.publicview..*Controller.*(..))")
+    public Object showExhibition(ProceedingJoinPoint jp) throws Throwable {
+        Object[] args = jp.getArgs();
+        MethodSignature signature = (MethodSignature) jp.getSignature();
+        int indexofModel = (Arrays.asList(signature.getParameterTypes())).indexOf(Model.class);
+        Exhibition exhibition = (Exhibition) exhibitionManager.getStartExhibition(); 
+        if(exhibition==null) {
+            return "home";
```

Comment:
that seems wrong, you probably want the actual controller take over from here so that the correct message is shown.

Reply:
If (exhibition == null) returns to controller, then controller loads slides even if no exhibition exists. Home shows that there is no exhibition at the moment.

Reply:
It should redirect to the start of the exhibition in that case. If we change something on the ’not setup yet’ screen, you don’t want to redo all of that in the aspect.